src/clj/rems/application/cleaner.clj

```diff
@@ -0,0 +1,57 @@
+(ns rems.application.cleaner
+  (:require [mount.core :as mount]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [rems.scheduler :as scheduler]
+            [rems.db.applications :as applications]
+            [rems.config :refer [env]]))
+
+(defn- is-expired-draft?
+  [application]
+  (let [expiration-threshold (:draft-application-expiration-threshold env)
+        last-activity (:application/last-activity application)
+        state (:application/state application)]
+    (and (= :application.state/draft state)
+         (time/before? last-activity (time/minus (time/now) (time/days expiration-threshold))))))
+
+(deftest test-is-expired-draft
+  (with-redefs [env {:draft-application-expiration-threshold 90}]
+    (testing "should identify expired draft application"
+      (are [expected input] (= expected (is-expired-draft? input))
+        true {:application/state :application.state/draft
+              :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))}
+        false {:application/state :application.state/draft
+               :application/last-activity (time/now)}
+        false {:application/state :application.state/submitted
+               :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))}))))
+
+(defn process-expired-draft-applications! []
```

Comment:
We could perhaps name this `process-applications!` as the namespace tells what we are doing here. Or even `remove-expired-applications!` which would be more concrete than "process" which could mean something other than deletion. When we possibly add email sending then perhaps the former is a better name.